x-pack/plugin/blob-cache/src/main/java/org/elasticsearch/blobcache/shared/SharedBytes.java

```diff
@@ -339,7 +345,12 @@ public int write(ByteBuffer src, int position) throws IOException {
             assert position % PAGE_SIZE == 0;
             assert src.remaining() % PAGE_SIZE == 0;
             checkOffsets(position, src.remaining());
-            int bytesWritten = fileChannel.write(src, pageStart + position);
+            int bytesWritten;
+            try {
+                bytesWritten = fileChannel.write(src, pageStart + position);
+            } catch (ClosedChannelException e) {
+                throw new AlreadyClosedException("Cache has been closed", e);
```

Comment:
Is this still going to be caught by SearchIndexInput which leads to fallback to reading from the object store?